lib/Conversion/ImportVerilog/Expression.cpp

```diff
@@ -267,6 +271,65 @@ struct ExprVisitor {
     return {};
   }
 
+  Value visit(const slang::ast::ConditionalExpression &expr) {
+    auto type = context.convertType(*expr.conditions.begin()->expr->type);
+    Value cond = convertToSimpleBitVector(
+        context.convertExpression(*expr.conditions.begin()->expr));
+    if (!cond)
+      return {};
+    if (!cond.getType().isa<mlir::IntegerType>()) {
+      auto zeroValue = builder.create<moore::ConstantOp>(loc, type, 0);
+      cond = builder.create<moore::InEqualityOp>(loc, cond, zeroValue);
+    }
```

Comment:
I think you want a boolean cast here instead. Check out the lowering of the `if` statement. I think that had to do a similar thing and that might be using the `bool_cast` op to capture this.

Reply:
> I think you want a boolean cast here instead. Check out the lowering of the `if` statement. I think that had to do a similar thing and that might be using the `bool_cast` op to capture this. 

@fabianschuiki 
Thanks for your suggestion :heart: ! However, when using `bool_cast` op, It throws out an error that `'scf.if' op operand #0 must be 1-bit signless integer, but got '!moore.bit'`. The reason why it failed is that the `condition` of `scf.if` needs `I1` type rather than `bit` type. Given this, I add `ConversionOp` to convert `bit` type to `I1` type and apply it to `scf.if` op. The specific code is as follows：
```cpp
  Value visit(const slang::ast::ConditionalExpression &expr) {
    Value cond = convertToSimpleBitVector(
        context.convertExpression(*expr.conditions.begin()->expr));
    if (!cond)
      return {};

    cond = builder.create<moore::BoolCastOp>(loc, cond);
    cond = builder.create<moore::ConversionOp>(loc, builder.getI1Type(), cond);

    auto ifOp = builder.create<mlir::scf::IfOp>(
        loc, cond,
        [&](OpBuilder &builder, Location loc) {
          builder.create<mlir::scf::YieldOp>(
              loc, context.convertExpression(expr.left()));
        },
        [&](OpBuilder &builder, Location loc) {
          builder.create<mlir::scf::YieldOp>(
              loc, context.convertExpression(expr.right()));
        });
    return ifOp.getResult(0);
  }
```
The results of `c = a ? a : b;` are as follows:
```
%0 = moore.gt %a, %b : !moore.int -> !moore.bit
%1 = moore.bool_cast %0 : !moore.bit -> !moore.bit
%2 = moore.conversion %1 : !moore.bit -> i1
      %3 = scf.if %2 -> (!moore.int) {
        scf.yield %a : !moore.int
      } else {
        scf.yield %b : !moore.int
      }
      moore.mir.bpassign %c, %3 : !moore.int
```
I would like to know if it is OK. Looking forward to your reply! :heart:

Reply:
Yeah I think this would definitely work! :smile: